processing/src/main/java/org/apache/druid/segment/incremental/OnheapIncrementalIndex.java

```diff
@@ -248,9 +268,14 @@ private void factorizeAggs(
   )
   {
     rowContainer.set(row);
-    for (int i = 0; i < metrics.length; i++) {
+    int aggLength = isPreserveExistingMetrics() ? aggs.length / 2 : aggs.length;
```

Comment:
You can use here `metrics.length` instead.

Reply:
Done